src/java/org/apache/cassandra/index/sai/plan/VectorTopKProcessor.java

```diff
@@ -96,7 +96,7 @@ public VectorTopKProcessor(ReadCommand command)
     public <U extends Unfiltered, R extends BaseRowIterator<U>, P extends BasePartitionIterator<R>> BasePartitionIterator<?> filter(P partitions)
     {
         // priority queue ordered by score in ascending order
-        PriorityQueue<Triple<PartitionInfo, Row, Float>> topK = new PriorityQueue<>(limit, Comparator.comparing(Triple::getRight));
+        PriorityQueue<Triple<PartitionInfo, Row, Float>> topK = new PriorityQueue<>(limit + 1, Comparator.comparing(Triple::getRight));
```

Comment:
worth a comment?

Reply:
I'll add one. Adding 1 to the limit prevents the PQ from needing to grow to 1.5x size when we have more keys that match the query than the limit.